### PR TITLE
fix(component-plugin-modernizer-tool) update GitHub URL following organization migration

### PR DESCRIPTION
**Pull request by dduportal**
# Link to GitHub repository

https://github.com/jenkins-infra/plugin-modernizer-tool

This PR updates the GitHub URL of the repository which has been migrated from `jenkinsci` to `jenkins-infra` GitHub organization as part of https://github.com/jenkins-infra/helpdesk/issues/4262

```[tasklist]
### Reviewer checklist
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@Wadeck`) in this pull request. If an email contact is changed, wait for approval from the security officer.
```
There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it.

### CODE REVIEW

**Comment by dduportal** [comment]
Weird: the checks are failing with an unrelated error (but the error is not present on the principal branch) :| Did I miss something?

**Comment by timja** [comment]
> Weird: the checks are failing with an unrelated error (but the error is not present on the principal branch) :| Did I miss something?

> Exception in thread "main" java.lang.Exception: Cannot have CD ('cd') enabled without specifying GitHub repository ('github'), for component: plugin-modernizer-tool

 its related

**Comment by dduportal** [comment]
Oh I see, thanks @timja !

File: src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionsUpdater.groovy [hunk 1]
```diff
@@ -151,7 +151,7 @@ class ArtifactoryPermissionsUpdater {
                     paths.addAll(definition.paths)
                 }
                 if (definition.cd && definition.getCd().enabled) {
-                    if (!definition.github.matches('(jenkinsci)/.+')) {
+                    if (!definition.github.matches('(jenkinsci|jenkins-infra)/.+')) {
                         throw new Exception("CD is only supported when the GitHub repository is in @jenkinsci")
```
Comment (daniel-beck) [inline src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionsUpdater.groovy:155]: This line should have been updated accordingly.

**Pull request merged**